server/protocols/jmap/src/test/java/org/apache/james/jmap/model/FilterConditionTest.java

```diff
@@ -122,4 +130,36 @@ public void buildShouldWork() {
     public void shouldRespectJavaBeanContract() {
         EqualsVerifier.forClass(FilterCondition.class).verify();
     }
+
+    @Test
+    public void buildShouldWorkWhenGivenHasKeywords() {
+        String hasKeyword = "$Draft AND $Flagged";
+
+        FilterCondition filterCondition = FilterCondition.builder()
+            .hasKeyword(Optional.of(hasKeyword))
+            .build();
+        assertThat(filterCondition.getHasKeyword().get()).isEqualTo(hasKeyword);
+    }
+
+    @Test
+    public void buildShouldNotCareAboutInvalidKeywordWhenGivenKeywordsWithUnknownValue() {
+        String hasKeyword = "$Draft AND NotValid";
```

Comment:
🔴  this is not doing what do you think it does!

hasKeyword and notKeyword input is a **Keyword**, thus is subject to Keyword limitations: no space no special characters, 255 char max.

Please correct these tests and add the missing unit tests here.